samcli/lib/docker/log_streamer.py

```diff
@@ -73,7 +74,12 @@ def _stream_write(self, _id: str, status: str, stream: str, progress: str, error
         :param error: docker log error
         """
         if error:
-            raise LogStreamError(msg=error)
+            if self._throw_on_error:
+                raise LogStreamError(msg=error)
+            else:
+                self._stream.write_str(error)
+                return
```

Comment:
When we write this to the error stream, shouldn't we also throw it again? Changing this variable into opposite boolean to indicate whether an error should be written into stream or not could be better 🤔

Reply:
No, because you might have multiple error lines. Also the throw in this features case is with the whole Docker build exception as the content vs just an individual error line.